sqlmesh/utils/pydantic.py

```diff
@@ -10,7 +10,7 @@ class PydanticModel(BaseModel):
     class Config:
         arbitrary_types_allowed = True
         extra = "forbid"
-        json_encoders = {exp.Expression: lambda e: e.sql(dialect=e.meta.get("dialect"))}
+        json_encoders = {exp.Expression: lambda e: e.meta.get("sql") or e.sql()}
```

Comment:
why would we ever fall back to e.sql()?

Reply:
Yes, I plan to get rid of this soon -- left over due to testing

Reply:
@tobymao changed to only do `e.meta["sql"]` but it seems like it's gonna need a bit of work to fix tests & ensure we're adding the `"sql"` key to all expressions (e.g. in model meta fields, etc). If the direction looks good I should be able to complete this tomorrow.

Reply:
what tests is this a problem?

Reply:
Several are failing with a `KeyError` due to the `"sql"` lookup but in many cases it might be because we have `from sqlglot import parse, parse_one` instead of `from sqlmesh.core.dialect import parse, parse_one`, and so the sqlmesh-specific logic might not be kicking off (already fixed some tests in test_state_sync with just this change).

Take a look at Circle CI's report to get a feel if you want.

Reply:
Brought it back in after all because python and seed models don't have raw sql associated with them and so we need to fallback to `e.sql()` for them.